pkg/ebpf/c/common/filtering.h

```diff
@@ -357,6 +357,9 @@ statfunc u64 match_data_filters(program_data_t *p, u8 index)
     u64 explicit_disable_policies = 0;
     u64 explicit_enable_policies = 0;
     u64 default_enable_policies = 0;
+    // Determine policies that do not use any type of string filter (exact, prefix, suffix)
+    u64 mask_no_str_filter_policies =
+        ~str_filter->exact_enabled & ~str_filter->prefix_enabled & ~str_filter->suffix_enabled;
```

Comment:
In the future, this could potentially be precomputed in userspace. Does the string_filter_config_t struct have any unused padding that could accommodate it? If so, precomputing this could also optimize the check at line 353. That flag could represent either enabled or disabled states, effectively encapsulating both scenarios. This would streamline the logic by providing a single source of truth for the checks.

Reply:
Good point, @geyslan. I think this optimization is worth it. Precomputing it in user space is a good idea. @yanivagman also suggested that maybe we should precompute the bitmaps in user-space (we need to do some PoCs to see if its possible). I think we can make an effort in the near future to enumerate and handle those places. WDYT?